### PR TITLE
Pull request for Feature-Integration-Specs branch

### PR DESCRIPTION
**Pull request by chuaindia**
In this  PR, we incorporated the following: 

- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
- Integration specs
      Capybara is used to write integration tests for each view in your project.

- In the User index page followings are available :
      - [ ] one can see the username of all other users.
      - [ ] one can see the profile picture for each user.
      - [ ] one can see the number of posts each user has written.
      - [ ] When one user clicks on a user, the user's show page will be invoked.

- In the user show page followings are available:
     - [ ] one can see the user's profile picture.
     - [ ] one can see the user's username.
     - [ ] one can see the number of posts the user has written.
     - [ ] one can see the user's bio.
     - [ ] one can see the user's first 3 posts.
     - [ ] one can see a button that views a user's posts.
     - [ ] When one clicks a user's post, it redirects me to that post's show page.
     - [ ] When one clicks to see all posts, it redirects me to the user's post's index page.

- In the user post index page followings are available :

    - [ ] one can see the user's profile picture.
    - [ ] one can see the user's username.
    - [ ] one can see the number of posts the user has written.
    - [ ] one can see a post's title.
    - [ ] one can see some of the post's body.
    - [ ] one can see the first comments on a post.
    - [ ] one can see how many comments a post has.
    - [ ] one can see how many likes a post has.
    - [ ] one can see a section for pagination if there are more posts than fit on the view.
    - [ ] When I click on a post, it redirects me to that post's show page.

In the user post show page followings are available :

   - [ ] one can see the post's title.
   - [ ] one can see who wrote the post.
   - [ ] one can see how many comments it has.
   - [ ] one can see how many likes it has.
   - [ ] one can see the post body.
   - [ ] one can see the username of each commenter.
   - [ ] one can see the comment each commenter left.

Before_screenshot

After_screenshot

### CODE REVIEW

File: spec/views/posts/user_post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    end
+    it 'should show how many comments a post has' do
+      post = @user.posts.first
+      expect(page).to have_content("Comments: #{post.comments_counter}")
+    end
+    it 'should have button for pagination' do
+      expect(page).to have_button('Pagination')
+    end
+  end
+end
```
Comment (Nemwel-Boniface) [inline spec/views/posts/user_post_index_spec.rb]: - [x] You seem to have forgotten to add the tests for "I can see how many likes a post has." Kindly include it.

image

- [x] You seem to have forgotten to add the tests for "When I click on a post, it redirects me to that post's show page." Kindly include it.

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @chuaindia        :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

- [x] I  have double-checked and I was not able to see the tests for the posts show page as it is part of the project requirements, kindly include them.

image

- [x] For some reason, when I run ```rspec spec/views``` a majority of the tests are failing. Kindly look into why this is the case.

image



_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/posts/user_post_index_spec.rb (hunk 1 above)
Comment (chuaindia) [reply spec/views/posts/user_post_index_spec.rb]: In my machine rspec spec/views are not showing any error. I've attached the screenshot here.

Screenshot (218)

Comment (chuaindia) [reply spec/views/posts/user_post_index_spec.rb]: Updated everything following Reviewer's review. Screenshots are also attached:


Screenshot (219)

**Review by DammyShittu — APPROVED** [review]
Hi @chuaindia and @samlorlah,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**